packages/frontend-2/middleware/canViewProjectTokens.ts

```diff
@@ -0,0 +1,41 @@
+import { graphql } from '~/lib/common/generated/gql'
+import { useApolloClientFromNuxt } from '~~/lib/common/composables/graphql'
+import { convertThrowIntoFetchResult } from '~~/lib/common/helpers/graphql'
+import { projectRoute } from '~~/lib/common/helpers/route'
+
+const canViewProjectWebhooksQuery = graphql(`
+  query CanViewProjectWebhooks($projectId: String!) {
+    project(id: $projectId) {
+      id
+      permissions {
+        canReadWebhooks {
+          ...FullPermissionCheckResult
+        }
+      }
+    }
+  }
+`)
+
+/**
+ * Apply this to a page to prevent unauthenticated access to tokens and ensure the user is the owner
+ */
+export default defineNuxtRouteMiddleware(async (to) => {
+  const client = useApolloClientFromNuxt()
+
+  const projectId = to.params.id as string
+  const { data } = await client
+    .query({
+      query: canViewProjectWebhooksQuery,
```

Comment:
huh, the page is mixing up the concept of webhooks and tokens

Reply:
Ah yeah good catch, this was a left over from when this policy didn't exist yet and I forgot to update it 😅 Will change!